src/test/java/picard/sam/FastqToSamTest.java

```diff
@@ -177,6 +181,67 @@ public void testEmptyFastqAllowed() throws IOException {
         convertFile(emptyFastq, null, FastqQualityFormat.Illumina, false, false, true);
     }
 
+    @Test
+    public void testStreamInputWithoutQuality() throws IOException {
+        ByteArrayOutputStream stderrStream = new ByteArrayOutputStream();
+        PrintStream newStderr = new PrintStream(stderrStream);
+        PrintStream oldStderr = System.err;
+        System.setErr(newStderr);
+
+        final File tmpFile = File.createTempFile("empty", ".sam");
+
+        final String[] args = {
+                "FASTQ=/dev/stdin",
+                "SAMPLE_NAME=sample001",
+                "OUTPUT=" + tmpFile
+        };
+
+        Assert.assertEquals(runPicardCommandLine(args), 1);
+        System.setErr(oldStderr);
+        Assert.assertTrue(stderrStream.toString().endsWith("QUALITY_FORMAT must be specified when input is not a regular file\n"));
+    }
+
+    @Test
+    public void testStreamInput() throws IOException {
```

Comment:
invoking bash/java this way seems brittle as anything but as I'm sure you've seen we're already doing it in `PipedDataTest` and I'm not really sure what the better solution is right now.

Reply:
Heh. Agreed... :grin: I originally tried redirecting `System.in`  from a `ByteArrayInputStream` containing the FASTQ string bytes and calling `runPicardCommandLine` in analogy to the test above, but that didn't work: the input was empty.